modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/classloader/ArtifactClassLoaderResolver.java

```diff
@@ -145,6 +147,42 @@ MuleDeployableArtifactClassLoader createApplicationClassLoader(ApplicationDescri
                                                                  Supplier<ArtifactClassLoader> domainClassLoader,
                                                                  PluginClassLoaderResolver pluginClassLoaderResolver);
 
+  /**
+   * Creates a class loader for an application. This will create the class loader itself and all of its internal required state:
+   * regionClassLoader, classLoaders for plugins.
+   * </p>
+   * The given descriptor must have its dependencies' exported packages sanitized (i.e. dependencies must not have packages that
+   * are already exported by their transitive dependencies nor any other dependency).
+   *
+   * @param descriptor                the descriptor of the application to generate a class loader for.
+   * @param pluginClassLoaderResolver allows the user to provide a class loader for the given plugin, otherwise it will be
+   *                                  created.
+   * @param urls                      a list of {@link URL} pointing to external resources
```

Comment:
classes will be loaded from those urls, not just resources, right?

Reply:
Yes, I will correct the jdoc